src/infer_request.cc

```diff
@@ -400,6 +400,23 @@ InferRequest::DeleteResponseFactory()
 #endif
 
 #ifdef TRITON_PB_STUB
+bool
+InferRequest::IsCancelled()
+{
+  std::unique_ptr<Stub>& stub = Stub::GetOrCreateInstance();
+  if (!stub->StubToParentServiceActive()) {
+    LOG_ERROR << "Cannot communicate with parent service";
+    return false;
+  }
+  if (request_address_ == 0) {
+    LOG_ERROR << "Request address not provided (default initialized?)";
```

Comment:
Instead of logging the error you can throw an exception.

Reply:
Sure. The logic is moved to "python_be.cc" after some changes on decoupled case.